src/core.mjs

```diff
@@ -0,0 +1,273 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {inspect} from 'node:util'
+import {spawn} from 'node:child_process'
+import {chalk, psTree, which} from './goods.mjs'
+import {boundCtxKey, getCtx, runInCtx, setRootCtx} from './als.mjs'
+import {randId} from './util.mjs'
+import {printStd, printCmd} from './print.mjs'
+import {formatCmd, quote} from './guards.mjs'
+
+export { getCtx, runInCtx, boundCtxKey }
+
+export function $(...args) {
+  let resolve, reject
+  let promise = new ProcessPromise((...args) => [resolve, reject] = args)
+
+  promise[boundCtxKey] = {
+    ...getCtx(),
+    id:      randId(),
```

Comment:
Why do we need an id?

Reply:
For debugging in most. But mb it's redundant. async_hook produces smth similar too. I'll remove